### DISCUSSION TITLE
scale_model_input in schedulers

### DISCUSSION THREAD

**Discussion by aradhyamathur** [huggingface/diffusers#7702] [General]
Hi, I was going over the scale_model_input implementation in the schedulers, I did not find any scaling taking place. Maybe I missed the part where the input gets scaled using the timestep. 
https://github.com/huggingface/diffusers/blob/6b02323a602a66841729c3a5d60844b24aa81ff2/src/diffusers/schedulers/scheduling_ddim.py#L171 
Can someone guide as to how the scaling takes place since the implementation just returns the sample back in the implementation attached above

**Comment by tolgacangoz** [comment]
You can [blame it](https://github.com/huggingface/diffusers/blame/6b02323a602a66841729c3a5d60844b24aa81ff2/src/diffusers/schedulers/scheduling_ddim.py#L171), go to its PR, and try to understand the rationale behind it.

**Comment by wzic** [comment]
same question. I have tried to print the difference between the before and the after sample. The difference is 0. It seems that the function just does nothing.

**Reply by asomoza** [reply]
@tolgacangoz already answered it but just so we can have the reason here and maybe prevent being asked the same question again, the docstring pretty much answers it and also if you trace the PR that added it:

docstring:

> Ensures interchangeability with schedulers that need to scale the denoising model input depending on the
        current timestep.

PR:

> Every scheduler needs to implement the scale_model_input(sample, timestep) (even if it just returns the sample) that scales the denoising model's input based on the current timestep. The method should be called before every model() call.

**Reply by AnsonSavage** [reply]
Maybe this is an uninformed question, but in that case, is there any reason it wasn't designed to just use the template method pattern (or otherwise) to call `scale_model_input()` at the beginning of every `model()` call so that the caller didn't need to worry about it?
